app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt

```diff
@@ -131,17 +142,18 @@ class MainActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListene
     private fun exportAsCsv() {
         launch {
             val content = buildCsv()
-
+            val filename = "prepaid-balance-${System.currentTimeMillis()}.csv"
             try {
-                val filename = "prepaid-balance-${System.currentTimeMillis()}.csv"
                 writeToFileInDownloads(content, filename)
                 showSnackbar(getString(R.string.export_saved_file, filename))
                 return@launch
-            } catch (e: Exception) {
-                Log.e(TAG, "Error saving file", e)
+            } catch (e: FileNotFoundException) {
+                if (hasPermissions(READ_EXTERNAL_STORAGE) && hasPermissions(WRITE_EXTERNAL_STORAGE)) {
+                    exportAsCsv()
```

Comment:
When the export fails due to any error except permissions, this creates an endless loop.
I'd catch `SecurityException` here and then launch the permission request. Also catch `Exception` and show the error snackbar (as before).

Reply:
I could add another catch exception to handle any 

> Please push to the same branch, so it appears in this PR.

I will delete this branch and create a new branch